src/components/OcNotificationMessage.spec.js

```diff
@@ -0,0 +1,86 @@
+import OcNotificationMessage from "./OcNotificationMessage.vue"
+import { shallowMount } from "@vue/test-utils"
+jest.useFakeTimers()
+
+describe("OcNotificationMessage", () => {
+  function getWrapper(props = {}) {
+    return shallowMount(OcNotificationMessage, {
+      propsData: {
+        ...props,
+        title: "Test passed",
+      },
+    })
+  }
+  const selectors = {
+    messageTitle: ".oc-notification-message-title",
+    messageContent: ".oc-notification-message-content",
+    messageWrapper: ".oc-alert div",
+  }
+  describe("title prop", () => {
+    it("should set notification message title", () => {
+      const wrapper = getWrapper()
+      expect(wrapper.find(selectors.messageTitle).text()).toBe("Test passed")
+    })
+  })
+  describe("message prop", () => {
+    it("should render message, if message is provided", () => {
+      const wrapper = getWrapper({ message: "This is a test notification message" })
+      const messageEl = wrapper.find(selectors.messageContent)
+      expect(messageEl.exists()).toBeTruthy()
+      expect(messageEl.text()).toBe("This is a test notification message")
+    })
+    it("should not render message, if message is not provided", () => {
+      const wrapper = getWrapper({})
+      const messageEl = wrapper.find(selectors.messageContent)
+      expect(messageEl.exists()).toBeFalsy()
+    })
+  })
+  describe("status prop", () => {
+    it("should not allow values other than passive, primary, success, warning, danger", () => {
+      try {
+        getWrapper({
+          status: "not-valid",
+        })
+        throw new Error(`Provided value for prop "status" is valid.`)
+      } catch (e) {
+        /* eslint-disable-next-line jest/no-conditional-expect, jest/no-try-expect */
+        expect(e).toContain(
+          '[Vue warn]: Invalid prop: custom validator check failed for prop "status".'
+        )
+      }
+    })
+    it.each(["passive", "primary", "success", "warning", "danger"])(
+      "it should set provided status as class for wrapper",
+      status => {
+        const wrapper = getWrapper({ status: status })
+        expect(wrapper.attributes("class")).toContain(`uk-notification-message-${status}`)
+      }
+    )
+    it("should set status as icon variation", () => {
+      const wrapper = getWrapper({ status: "primary" })
+      const iconElement = wrapper.find("oc-icon-stub")
+      expect(iconElement.exists()).toBeTruthy()
+      expect(iconElement.attributes("variation")).toBe("primary")
+    })
+    describe("role and aria live of message content wrapper", () => {
+      it("if status is not danger", () => {
+        const wrapper = getWrapper({})
+        const messageContentEl = wrapper.find(selectors.messageWrapper)
+        expect(messageContentEl.attributes("role")).toBe("status")
+        expect(messageContentEl.attributes("aria-live")).toBe("polite")
+      })
+      it("if status is danger", () => {
```

Comment:
I am kinda missing here what the step should actually do. If true then what?

Reply:
fixed